src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -515,6 +520,16 @@ class SavePlanFile extends Action {
 	}
 
 	public override async run(context: QueryPlan2): Promise<void> {
+		const saveFileUri = await context.fileDialogService.showSaveDialog({
```

Comment:
1. you should give a default file name
2. need to handle the save dialog canceled scenario.